tasty-dejafu/Test/Tasty/DejaFu.hs

```diff
@@ -146,37 +146,43 @@ instance IsOption Way where
 -- | Automatically test a computation. In particular, look for
 -- deadlocks, uncaught exceptions, and multiple return values.
 --
--- @since 2.0.0.0
+-- @since unreleased
 testAuto :: (Eq a, Show a)
-  => Program pty IO a
+  => TestName
+  -- ^ The name of the test group.
+  -> Program pty IO a
   -- ^ The computation to test.
   -> TestTree
 testAuto = testAutoWithSettings defaultSettings
 
 -- | Variant of 'testAuto' which tests a computation under a given
 -- execution way and memory model.
 --
--- @since 2.0.0.0
+-- @since unreleased
 testAutoWay :: (Eq a, Show a)
   => Way
   -- ^ How to execute the concurrent program.
   -> MemType
   -- ^ The memory model to use for non-synchronised @IORef@ operations.
+  -> TestName
+  -- ^ The name of the test group.
   -> Program pty IO a
   -- ^ The computation to test.
   -> TestTree
 testAutoWay way = testAutoWithSettings . fromWayAndMemType way
 
 -- | Variant of 'testAuto' which takes a settings record.
 --
--- @since 2.0.0.0
+-- @since unreleased
 testAutoWithSettings :: (Eq a, Show a)
   => Settings IO a
   -- ^ The SCT settings.
+  -> TestName
+  -- ^ The name of the test group.
   -> Program pty IO a
   -- ^ The computation to test.
   -> TestTree
-testAutoWithSettings settings = testDejafusWithSettings settings
+testAutoWithSettings settings groupName = testDejafusWithSettings settings groupName
```

Comment:
I’d expect the test name to be first. Looks better in tasty code, a bit like named bullet points.

But maybe not a big deal, one probably defines a local helper anyways if one often has to set settings.

Reply:
Yeah, I was kind of torn on this, but this matches the argument order for the functions which only do a single test (`settings -> test name -> test predicate -> test program`), and I figured it's probably better to be consistently awkward.

I'll get this merged and on hackage as tasty-dejafu-2.1.0.0.